docs/source/single_page_run_through.rst

```diff
@@ -0,0 +1,400 @@
+`x` and `y`, one of which will be kept constant whilst the other is varied to find the root.
+
+When `newton_raphson` is called, it takes the `root=my_func` kwarg, and inspects the signature
+of `my_func`. It finds that my_func takes 2 arguments `x` and `y`, and expects the first argument
+(`x` in this case) to be the parameter that is modified by the algorithm to find the root.
+
+As a result, we need to be able to pass `y` from the top level caller all the way down to the
+places where 'root' is called. For example, line 5 in the example above needs to be changed to::
+
+    new_y = my_func(next_x, y)
+
+In order to do this, the `newton_raphson` function needs its signature modified to take `y`
+as an argument. This is what `fastats` performs: at any level in the AST, `fastats` will modify
+the function signatures and ensure that the correct arguments are passed to all functions,
+in order to allow any function to be modified by passing it as a keyword argument at the
+top-level.
+
+In this example, `deriv` will numerically calculate the derivative at each point, which is the
+reason for requiring nested function substitution from the AST transform. It is possible to use
+the same `deriv` function for a wide range of `root` functions, as the numerical derivative just
+calls the `root` function with the x-values bumped in either direction. As a result, we need the
+`root=` keyword argument to replace all values in the child functions (including `deriv`), not
+just the one at the top-level. This allows us to have a very efficient and easy-to-learn API
+without sacrificing performance.
+
+However, in this case we can trivially calculate the analytical derivative by hand, and replace the `deriv` function
+like this::
+
+    def my_deriv(x, y):
+        return 2 * x + 2 * y
+
+    my_solve = newton_raphson(, 0.001, root=my_func, deriv=my_deriv)
+
+    my_solve(3, 4)
+
+Which allows us to **optionally** pass an optimised function, but fall back on a non-optimised
+version for experimenting/research.
+
+This is not-limited to specific functions - if you are happy with lower-precision in certain
+calculations, you can pass faster (lower-precision) versions of any mathematical functions,
+and `fastats` will replace them throughout the entire AST before sending onto numba, without
+requiring you to modify any code.
+
+As an example, some calculations require the complementary error function `erfc` to be calculated.
+The accuracy (precision) of `erfc` depends partially on how many terms are in the multiplication. By
+reducing the number of terms we can speed up calculations at the expense of accuracy.
+
+If you are happy with 8 decimal places, you can speed up calculations by passing `erfc8`::
+
+    from fastats.core.erfc import erfc8
+
+    my_solve = newton_raphson(, 0.000, root=my_func, erfc=erfc8)
+
+To increate precision (at the expense of calculation time), you could use `erfc16`::
```

Comment:
increate seems like an odd choice here (and probably very weird for non-native speakers), consider using increase instead?

Reply:
👍 typo, nice one